app/DAO/BakeryDatabase.scala

```diff
@@ -68,41 +83,85 @@ class BakeryDatabase @Inject() (
     }
   }
 
-  def getProductById(id: String): Future[Option[String]] = {
-    Future {
-      db.withConnection { conn =>
-        try {
-          val statement = {
-            conn.prepareStatement("SELECT * FROM product WHERE id::text = ?")
-          }
-          statement.setString(1, id)
+  def getProductById(id: UUID): Future[Option[Product]] = {
+    sql"""SELECT * FROM product WHERE id = $id"""
+      .query[Product]
+      .option
+      .transact(bakeryTransactor.xa)
+      .unsafeToFuture()
+  }
 
-          println(statement.toString)
-          val selectResult = statement.executeQuery()
-          val result = if (selectResult.next()) {
-            Some(
-              "id: " + selectResult.getString(
-                "id"
-              ) + ", name: " + selectResult.getString(
-                "name"
-              ) + ", quantity: " + selectResult.getInt(
-                "quantity"
-              ) + ", price: " + selectResult.getDouble(
-                "price"
-              ) + ", created at: " + selectResult.getTimestamp(
-                "created_at"
-              ) + ", updated at: " + selectResult.getTimestamp(
-                "updated_at"
-              )
-            )
-          } else {
-            None
-          }
-          result
-        } catch {
-          case e: Exception => Option(e.printStackTrace().toString)
-        }
-      }
+  def createProduct(newProduct: ProductUpdateRequest): Int = {
+
+    /** Steps for creating a record
+      *  1. bring in the json, everything should be provided except for the uuid
+      *  2. use query.update to commit the record
+      *  3. run the transaction
+      *  4. return a variable the controller can check to know the transaction succeeded (201 Created)
+      */
+    val newStamp = OffsetDateTime.now()
+    (newProduct.name, newProduct.quantity, newProduct.price) match {
+      case (Some(name), Some(quantity), Some(price)) =>
+        sql"""INSERT INTO product VALUES (gen_random_uuid(), $name, $quantity, $price, $newStamp, $newStamp)""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case _ => 0
+    }
+  }
+
+  def updateProduct(id: UUID, changesProduct: ProductUpdateRequest): Int = {
+
+    /** Steps for updating a record
+      *  1. bring in the json, json should be the values to change (8 combinations):
+      *     - none, name, qty, price, name+qty, name+price, qty+price, name+qty+price
+      *  2. check what values are given to update, form query on these params
+      *  3. query with .update on the given id
+      *  4. run the transaction
+      */
+    val newName = changesProduct.name
+    val newQty = changesProduct.quantity
+    val newPrice = changesProduct.price
```

Comment:
superfluous

Reply:
rewrote values into the pattern matching

Reply:
Yeah I guess you can do that. In general you could have also pattern matched on 

```
(changesProduct.name, changesProduct.quantity, changesProduct.price) match { // etc
```